onnx/onnx-ml.proto

```diff
@@ -234,6 +234,70 @@ message NodeProto {
 
   // Named metadata values; keys should be distinct.
   repeated StringStringEntryProto metadata_props = 9;
+
+  // Configuration of multi-device annotations.
+  repeated NodeDeviceConfigurationProto configuration = 10;
+}
+
+// IntIntListEntryProto follows the pattern for cross-proto-version maps.
+// See https://developers.google.com/protocol-buffers/docs/proto3#maps
+message IntIntListEntryProto {
+  optional int64 key = 1;
+  repeated int64 value = 2;
+};
+
+// Multi-device configuration proto for NodeProto.
+message NodeDeviceConfigurationProto {
+    // ID of the configuration.
+    optional string configuration_id = 1;
```

Comment:
How is this field used? Is it required by the spec? For optional fields (aka most of the fields) I think we should still specify in comments whether they are required in practice, even though we do want optional for proto definition.

Reply:
Prefixed `(Required)` in the comments describing the optional fields that are required. Let me know if there's a different way of doing it.

Reply:
I suggest using the "MUST", "SHOULD", "MAY" wording to be consistent with the rest of the spec: https://github.com/onnx/onnx/blob/main/docs/IR.md#open-neural-network-exchange-intermediate-representation-onnx-ir-specification

Reply:
e.g. "MUST be specified"

Reply:
Examples in the proto file: see [here](https://github.com/onnx/onnx/blob/c4031757d77487dd59bad35ad7a596413903ef07/onnx/onnx.in.proto#L150) ... "// The name field MUST be present for this version of the IR."

Reply:
Renamed `(Required)` -> `// This field MUST be present for this version of the IR.`